src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -323,8 +320,12 @@ internal ICollection<User> GetUsers(string clientId)
                 IDictionary<string, User> allUsers = new Dictionary<string, User>();
                 foreach (RefreshTokenCacheItem item in tokenCacheItems)
                 {
-                    User user = new User(item.User);
-                    allUsers[item.HomeObjectId] = user;
+                    if (environment.Equals(
+                        item.Environment, StringComparison.OrdinalIgnoreCase))
```

Comment:
It would be better to create a standard helper function _AreEnvironmentsEqual(env1, env2)_ that does the comparison. That avoids the risk of different _StringComparison_ kinds being used in different places.
The same applies to the comparison of _ClientId_.